### PR TITLE
Add Role Endpoints & Integration Tests

### PR DESCRIPTION
**Pull request by Vimbayinashe**
Closes issue #33 

This adds role endpoints and integration tests for RoleController.
ModelMapperConfig and IdNotFoundException are the same as in pull request #48

### CODE REVIEW

File: src/main/java/se/iths/librarysystem/repository/RoleRepository.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package se.iths.librarysystem.repository;
+
+import org.springframework.data.repository.CrudRepository;
+import se.iths.librarysystem.entity.RoleEntity;
+
+public interface RoleRepository extends CrudRepository<RoleEntity, Long> {
+    RoleEntity findByRole(String role);
```
Comment (ToniKaru) [inline src/main/java/se/iths/librarysystem/repository/RoleRepository.java:7]: I can't find where this is used. Add method call or remove.

Comment (Vimbayinashe) [reply src/main/java/se/iths/librarysystem/repository/RoleRepository.java:7]: It's used in UserService and DefaultRolesConfig.

**Review by ToniKaru — COMMENTED** [review]
I'm probably missing something, but can we remove PersonEntity and PersonEntityTest?

File: src/main/java/se/iths/librarysystem/entity/BookFormatEntity.java [hunk 2]
```diff
@@ -42,14 +42,14 @@ public BookFormatEntity(String formatName, boolean digital, int pageCount, Strin
...
-
     public Long getId() {
         return id;
     }
 
+    public void setId(Long id) {
+        this.id = id;
+    }
+
     public String getFormatName() {
```
[the remark this answers is not in the document]
Comment (ToniKaru) [reply src/main/java/se/iths/librarysystem/entity/BookFormatEntity.java:53]: And damn! This was a lot of great work!

**Comment by Vimbayinashe** [comment]
> I'm probably missing something, but can we remove PersonEntity and PersonEntityTest?

It has been removed.

File: src/main/java/se/iths/librarysystem/entity/BookFormatEntity.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Vimbayinashe) [reply src/main/java/se/iths/librarysystem/entity/BookFormatEntity.java]: Thank-you

**Review by ToniKaru — APPROVED** [review]
Looks good

**Pull request merged**